src/lib.rs

```diff
@@ -30,6 +30,25 @@ pub enum SpiError {
     Impl(ImplError),
 }
 
+/// A Serial specific error.
+///
+/// This error type contains errors specific to Serial peripherals. Also it has an `Impl` kind to pass
+/// through implementation specific errors occurring while trying to use a Serial peripheral.
+#[non_exhaustive]
+pub enum SerialError {
+    /// The peripheral receive buffer was overrun.
+    Overrun,
+    /// Received data does not conform to the peripheral configuration.
+    /// Can be caused by a misconfigured device on either end of the serial line.
+    FrameFormat,
+    /// Parity check failed.
+    Parity,
+    /// Serial line is too noisy to read valid data.
+    Noisy,
```

Comment:
Maybe we should call this `Noise` instead?

Reply:
Looks like I transposed that to make it consistent with `Parity`, though that makes it inconsistent with the rest of the crate.
(its also inconsistent with the upstream material I used as a reference).

Will correct.